subsys/bluetooth/services/dis.c

```diff
@@ -38,6 +47,54 @@ static struct dis_pnp dis_pnp_id = {
 };
 #endif
 
+#if defined(CONFIG_BT_GATT_DIS_SETTINGS)
+#define DIS_STR_MAX
+static u8_t dis_model[CONFIG_BT_GATT_DIS_STR_MAX] =
+	CONFIG_BT_GATT_DIS_MODEL;
+static u8_t dis_manuf[CONFIG_BT_GATT_DIS_STR_MAX] =
+	CONFIG_BT_GATT_DIS_MANUF;
+#if defined(CONFIG_BT_GATT_DIS_SERIAL_NUMBER)
+static u8_t dis_serial_number[CONFIG_BT_GATT_DIS_STR_MAX] =
+	CONFIG_BT_GATT_DIS_SERIAL_NUMBER_STR;
+#endif
+#if defined(CONFIG_BT_GATT_DIS_FIRMWARE_REVISION)
+static u8_t dis_firmware_revision[CONFIG_BT_GATT_DIS_STR_MAX] =
+	CONFIG_BT_GATT_DIS_FIRMWARE_REVISION_STR;
+#endif
+#if defined(CONFIG_BT_GATT_DIS_HARDWARE_REVISION)
+static u8_t dis_hardware_revision[CONFIG_BT_GATT_DIS_STR_MAX] =
+	CONFIG_BT_GATT_DIS_HARDWARE_REVISION_STR;
+#endif
+#if defined(CONFIG_BT_GATT_DIS_SOFTWARE_REVISION)
+static u8_t dis_software_revision[CONFIG_BT_GATT_DIS_STR_MAX] =
+	CONFIG_BT_GATT_DIS_SOFTWARE_REVISION_STR;
+#endif
+
+#define BT_GATT_DIS_MODEL_REF			dis_model
+#define BT_GATT_DIS_MANUF_REF			dis_manuf
+#define BT_GATT_DIS_SERIAL_NUMBER_STR_REF	dis_serial_number
+#define BT_GATT_DIS_FIRMWARE_REVISION_STR_REF	dis_firmware_revision
+#define BT_GATT_DIS_HARDWARE_REVISION_STR_REF	dis_hardware_revision
+#define BT_GATT_DIS_SOFTWARE_REVISION_STR_REF	dis_software_revision
+
+#else /* CONFIG_BT_GATT_DIS_SETTINGS */
+
+#define BT_GATT_DIS_MODEL_REF			\
+		CONFIG_BT_GATT_DIS_MODEL
```

Comment:
I wonder if we should cast to `const` to make sure those strings are stored in ROM and not RAM

Reply:
Not sure what to do here. You cannot just put the string into a const variable as there is no guarantee that you can reference a const variable in ROM as a pointer(depending on compiler/processor).